### PR TITLE
Upgrade Golang and Openshift module

### PR DESCRIPTION
**Pull request by jwsui**
Bump golang version to 1.19 to sync with Antrea and fix some golang runtime CVEs. And upgrade Openshift related module to fix CVEs imported by these modules(and their dependencies).

### CODE REVIEW

**Comment by ksamoray** [comment]
Change LGTM, can you check the build failure? Seems like `make generate` fails even though it depends on `make controller-gen`

**Comment by jwsui** [comment]
> Change LGTM, can you check the build failure? Seems like `make generate` fails even though it depends on `make controller-gen`
```
Starting in Go 1.17, installing executables with go get is deprecated. go install may be used instead.
```
Thanks, I thinks this is the reason, so use `go install` instead of `go get` after upgrading to go 1.19.

File: .golangci.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ linters:
   enable:
     - misspell
     - gofmt
-    - deadcode
+    - unused
```
Comment (jwsui) [inline .golangci.yml:18]: Upgrading golangci-lint to 1.51.0, and use `unused` instead as `deadcode` is deprecated since v1.49.0.

**Comment by ksamoray** [comment]
Note that role.yaml files under /deploy are auto-generated by make targets, as roles are mostly inherited from Antrea resources. Therefore changes to these files will be overwritten.
Roles which are only required by the operator should be placed in config/rbac/role_base.yaml (or role_base_ocp.yaml if they're specific to OpenShift).
See doc/release-process.md for details.

**Comment by jwsui** [comment]
> Note that role.yaml files under /deploy are auto-generated by make targets, as roles are mostly inherited from Antrea resources. Therefore changes to these files will be overwritten. Roles which are only required by the operator should be placed in config/rbac/role_base.yaml (or role_base_ocp.yaml if they're specific to OpenShift). See doc/release-process.md for details.

`patch` is required by [apply.ApplyObject](https://github.com/vmware/antrea-operator-for-kubernetes/pull/77/files#diff-136d6a5376439439f8d9596b5f2a0a3714a3189247d6fadede4797a96605c746R123) for every object from `antrea.yaml`. So I update `generate-role-yaml.py` to add verb `patch` to each object and changes under `/deploy` won't be overwritten as `generate-role-yaml.py` is changed.

**Review by ksamoray — APPROVED** [review]
Thanks for clarifying. LGTM

**Pull request merged**